### PR TITLE
[WIP][DO_NOT_MERGE]force user to choose a builder image

### PR DESCRIPTION
**Pull request by bparees**

### CODE REVIEW

File: pkg/generate/app/cmd/newapp.go [hunk 1]
```diff
@@ -203,7 +205,15 @@ func (c *AppConfig) AddArguments(args []string) []string {
 func (c *AppConfig) individualSourceRepositories() (app.SourceRepositories, error) {
 	first := true
 	for _, s := range c.SourceRepositories {
-		if repo, ok := c.refBuilder.AddSourceRepository(s); ok && first {
+		var requestedImage string
+		if strings.Contains(s, "~") {
```
Comment (smarterclayton) [inline pkg/generate/app/cmd/newapp.go]: Source repositories should have ~ in them.  Why is this here?

File: pkg/generate/app/sourcelookup.go [hunk 2]
```diff
@@ -62,6 +62,12 @@ func (d dockerfileContents) Contents() string {
 
 // IsPossibleSourceRepository checks whether the provided string is a source repository or not
 func IsPossibleSourceRepository(s string) bool {
+	if strings.Contains(s, "~") {
+		segs := strings.SplitN(s, "~", 2)
```
Comment (smarterclayton) [inline pkg/generate/app/sourcelookup.go]: This is not something I would expect.

**Comment by bparees** [comment]
@smarterclayton please just ignore this PR for now, i created it because i wanted to check something, but the code is not in any sort of final state.

**Comment by smarterclayton** [comment]
Ok

On Oct 24, 2015, at 4:22 PM, Ben Parees <notifications@github.com> wrote:

@smarterclayton <https://github.com/smarterclayton> please just ignore this
PR for now, i created it because i wanted to check something, but the code
is not in any sort of final state.

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/5297#issuecomment-150847643>.
